kaitai/struct_test.go

```diff
@@ -0,0 +1,46 @@
+package kaitai
+
+import (
+	"bytes"
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+// Declare two struct simulating types generated by Kaitai Struct compiler
+type oneStruct struct {
+	one int
+	_io *Stream
+}
+
+func (s *oneStruct) Kaitai_IO() *Stream {
+	return s._io
+}
+
+type twoStruct struct {
+	two int
+	_io *Stream
+}
+
+func (s *twoStruct) Kaitai_IO() *Stream {
+	return s._io
+}
+
+func WorkWithStruct(s Struct, t *testing.T, expectedSize int) {
+	actualSize, err := s.Kaitai_IO().Size()
+	assert.Equal(t, actualSize, int64(expectedSize))
+	assert.Nil(t, err)
```

Comment:
I think `err == nil` should be asserted first - if it's not true for some reason, then `actualSize` shouldn't be accessed at all.

Reply:
Done!